sklearn/preprocessing/_encoders.py

```diff
@@ -326,6 +418,215 @@ def _compute_drop_idx(self):
+        category_counts : list of ndarrays
+            list of category counts
+
+        n_samples : int
+            number of samples
+        """
+        self.infrequent_indices_ = [
+            self._compute_infrequent_indicies(category_count, n_samples,
+                                              col_idx)
+            for col_idx, category_count in enumerate(category_counts)]
+
+        # compute mapping from default mapping to infrequent mapping
+        default_to_infrequent_mappings = []
+        largest_infreq_idxs = []
+
+        for category_count, infreq_idx in zip(category_counts,
+                                              self.infrequent_indices_):
+            # no infrequent categories
+            if infreq_idx is None:
+                default_to_infrequent_mappings.append(None)
+                largest_infreq_idxs.append(None)
+                continue
+
+            # infrequent indicies exist
+            mapping = np.empty_like(category_count, dtype=np.int)
+            n_cats = mapping.size
+            n_infrequent_cats = infreq_idx.size
+
+            n_frequent_cats = n_cats - n_infrequent_cats
+            mapping[infreq_idx] = n_frequent_cats
+
+            frequent_indices = np.setdiff1d(np.arange(n_cats), infreq_idx)
+            mapping[frequent_indices] = np.arange(n_frequent_cats)
+
+            default_to_infrequent_mappings.append(mapping)
+
+            # compute infrequent category with the largest cardinality
+            largest_infreq_idx = np.argmax(category_count[infreq_idx])
+            largest_infreq_idxs.append(infreq_idx[largest_infreq_idx])
+
+        self._default_to_infrequent_mappings = default_to_infrequent_mappings
+        self._largest_infreq_indices = largest_infreq_idxs
+
+    def _map_to_infrequent_categories(self, X_int):
+        """Map categories to infrequent categories.
+
+        Note this will replace the encoding in X_int
+
+        Parameters
+        ----------
+        X_int: ndarray of shape (n_samples, n_features)
+            integer encoded categories
+        """
+        if not self._infrequent_enabled:
+            return
+
+        for col_idx, mapping in enumerate(
+                self._default_to_infrequent_mappings):
+
+            if mapping is None:
+                continue
+            X_int[:, col_idx] = np.take(mapping, X_int[:, col_idx])
+
+    def _get_default_invalid_category(self, col_idx):
+        """Get default invalid category for column index during `_transform`.
+
+        This function is pasesd to `_transform` to set the invalid categories.
+        """
+        infrequent_idx = self.infrequent_indices_[col_idx]
+        return 0 if infrequent_idx is None else infrequent_idx[0]
+
+    def _process_valid_mask(self, valid_mask, col_idx):
+        """Process the valid mask during `_transform`
+
+        This function is passed to `_transform` to adjust the mask depending
+        on if the infrequent column exist or not.
+        """
```

Comment:
I think it would be helpful to the reader if you documented Parameters and Returns here

Reply:
The Parameters and Returns were added.